osquery/core/linux/cpu.cpp

```diff
@@ -0,0 +1,145 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed as defined on the LICENSE file found in the
+ *  root directory of this source tree.
+ */
+
+#include "osquery/core/linux/cpu.h"
+
+#include <osquery/core/conversions.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/io/detail/quoted_manip.hpp>
+
+#include <fstream>
+#include <functional>
+
+namespace osquery {
+namespace cpu {
+
+namespace {
+
+Expected<std::string, Error> readSysCpuFile(char const* path) {
+  auto fin = std::ifstream(path, std::ios_base::in | std::ios_base::binary);
+  if (fin.fail() || fin.bad()) {
+    return createError(Error::IOError, "No access to the system file ") << path;
+  }
+  auto data = std::string{};
+  fin >> data;
+  return data;
+}
+
+Expected<std::size_t, Error> decodeCpuNumber(const std::string& str) {
+  auto exp = tryTo<std::size_t>(str);
+  if (exp.isError()) {
+    return createError(Error::IncorrectRange, "", exp.takeError())
+           << "Incorrect CPU number representation " << boost::io::quoted(str);
+  }
+  return exp.take();
+}
+
+} // namespace
+
+Expected<Mask, Error> decodeMaskFromString(const std::string& encoded_str) {
+  auto mask = Mask{};
+  if (encoded_str.empty()) {
+    return mask;
+  }
+  using It = boost::split_iterator<std::string::const_iterator>;
+  for (auto it = boost::make_split_iterator(
```

Comment:
const ..& ?

Reply:
It is an iterator, it have to be copied and incremented at the every step of the loop.
However it is a `const_iterator`, that means our data is in safe ;)